spectator/distsummary.py

```diff
@@ -1,68 +1,42 @@
-from abc import ABCMeta, abstractmethod
-from future.utils import with_metaclass
+from spectator.id import MeterId
+from spectator.spectatord import SpectatorD
 
-from spectator.atomicnumber import AtomicNumber
 
+class DistributionSummary:
+    """The value tracks the distribution of events. It is similar to a Timer, but more general,
+    because the size does not have to be a period of time."""
 
-class AbstractDistributionSummary(with_metaclass(ABCMeta)):
+    meter_type = "d"
 
-    @abstractmethod
-    def record(self, amount):
-        pass
+    def __init__(self, meter_id: MeterId, spectatord: SpectatorD = SpectatorD()) -> None:
+        self.meterId = meter_id
+        self._spectatord = spectatord
 
-    @abstractmethod
-    def count(self):
-        pass
-
-    @abstractmethod
-    def total_amount(self):
-        pass
-
-    @abstractmethod
-    def _measure(self):
-        pass
-
-
-class NoopDistributionSummary(AbstractDistributionSummary):
-
-    def record(self, amount):
-        pass
+    def record(self, amount: float) -> None:
+        if amount >= 0:
+            self._spectatord.publish(self.meter_type, self.meterId, amount)
 
-    def count(self):
+    @staticmethod
+    def count() -> float:
+        """Avoid breaking the API."""
         return 0
 
-    def total_amount(self):
+    @staticmethod
+    def total_amount() -> float:
+        """Avoid breaking the API."""
         return 0
 
-    def _measure(self):
-        return {}
 
+class PercentileDistributionSummaryV2:
```

Comment:
Should this share a common base type with `DistributionSummary`?

Reply:
Fixed.